sky/packages/sky/lib/src/fn3/homogeneous_viewport.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:sky/rendering.dart';
+import 'package:sky/src/fn3/framework.dart';
+import 'package:sky/src/fn3/basic.dart';
+
+typedef List<Widget> ListBuilder(int startIndex, int count);
+
+class HomogeneousViewport extends RenderObjectWidget {
+  HomogeneousViewport({
+    Key key,
+    this.builder,
+    this.itemsWrap: false,
+    this.itemExtent, // required
+    this.itemCount, // optional, but you cannot shrink-wrap this class or otherwise use its intrinsic dimensions if you don't specify it
+    this.direction: ScrollDirection.vertical,
+    this.startOffset: 0.0
+  }) : super(key: key) {
+    assert(itemExtent != null);
+  }
+
+  final ListBuilder builder;
+  final bool itemsWrap;
+  final double itemExtent;
+  final int itemCount;
+  final ScrollDirection direction;
+  final double startOffset;
+
+  RenderObjectElement createElement() => new HomogeneousViewportElement(this);
+
+  // we don't pass constructor arguments to the RenderBlockViewport() because until
+  // we know our children, the constructor arguments we could give have no effect
+  RenderObject createRenderObject() => new RenderBlockViewport();
+
+  bool isLayoutDifferentThan(HomogeneousViewport oldWidget) {
+    return itemsWrap != oldWidget.itemsWrap ||
+           itemsWrap != oldWidget.itemsWrap ||
+           itemExtent != oldWidget.itemExtent ||
+           itemCount != oldWidget.itemCount ||
+           direction != oldWidget.direction ||
+           startOffset != oldWidget.startOffset;
+  }
+
+  // all the actual work is done in the element
+}
+
+class HomogeneousViewportElement extends RenderObjectElement {
+  HomogeneousViewportElement(HomogeneousViewport widget) : super(widget);
+
+  List<Element> _children = const <Element>[];
+  bool _layoutDirty = true;
+  int _layoutFirstIndex;
+  int _layoutItemCount;
+
+  HomogeneousViewport get widget => super.widget;
```

Comment:
Instead of this override, you can just inherit from RenderObjectElement<HomogeneousViewport>